src/app/icd/BUILD.gn

```diff
@@ -25,13 +25,15 @@ source_set("manager-srcs") {
     "ICDManager.h",
   ]
 
-  public_deps = [ "${chip_root}/src/lib/core" ]
+  public_deps = [
+    "${chip_root}/src/lib/core",
+    "${chip_root}/src/platform:platform",
+  ]
 }
 
-# ICD Server Configuration
-# All configurations necessary to the ICD Server featureset need to in this configuration
-config("server-config") {
-  defines = [ "CHIP_CONFIG_ENABLE_ICD_SERVER" ]
+# Used to enabled ICD related method gated by a build argument in test
+config("test-manager-config") {
+  defines = [ "CHIP_CONFIG_ENABLE_ICD_SERVER=1" ]
 }
```

Comment:
I think a batter location would be putting this config in the build.gn for tests.